shell/edit/resources.cattle.io.restore.vue

```diff
@@ -93,7 +96,14 @@ export default {
     },
 
     encryptionSecretNames() {
-      return this.allSecrets.filter((secret) => !!(secret.data || {})['encryption-provider-config.yaml'] && secret.metadata.namespace === this.chartNamespace && !secret.metadata?.state?.error).map((secret) => secret.metadata.name);
+      const filtered = this.allSecrets.filter(
```

Comment:
It seems like this logic is repeated and could be replaced by a function, which is easier to test.